### PR TITLE
i18n Support: PR 1 - Second Attempt

### PR DESCRIPTION
**Pull request by StandingPadAnimations**
This is a second attempt at the first PR for i18n support, using Python's [`gettext` module](https://docs.python.org/3/library/gettext.html) instead of a custom JSON protocol. This ended up being easier since the GNU developers have tons of utilities for automatic generation (such as `xgettext`)

As of making PR 1, a good chunk of the MCprep N panel is translatable, taking advantage of GNU Gettext (which is the standard for translation). Translators may use utilities like [Poedit](https://poedit.net) to aid in translation. In the future, we may also look towards [Weblate](https://weblate.org/), though that's not a priority right now.

For developers, what changes is that strings should now use `text=env._(str)` instead of `text=str`, in order to make it translatable. In addition, for every release, we need to refresh the base POT file. This is simple though:
```sh
xgettext -d base -o MCprep_addon/MCprep_resources/Languages/mcprep.pot "MCPREP_FILE" --keyword="env._
```

(Now here's the copy paste from #458)
> Let no Minecraft animator be confused by MCprep's UI simply because of the language they speak.
> \- Goal defined in #439 

This is the first of a series of PRs to add support for translating the MCprep UI. The current roadmap is the following:
- PR 1 - Basic infastructure and MCprep N panel
- PR 2 - First half of all MCprep operators
- PR 3 - Second half of all MCprep operators
- PR 4 (To be decided) - Dynamic menus such as rigs and assets

PR 1 will be merged for MCprep 3.6. While not explicitly defined as a blocker for the 3.6 release, it will be treated as such unless decided otherwise.

### CODE REVIEW

**Comment by StandingPadAnimations** [comment]
So far, with a basic test (using English), everything seems fine. We'll have to port the Chinese translation we've received in #458 and see how well that works though. No errors though (thank god)

**Comment by StandingPadAnimations** [comment]
As I try to salvage what we had on the old branch, I think I need to do some of the previous ID stuff again, cause this is getting out of hand
image

**Comment by StandingPadAnimations** [comment]
The only changes that should be required though is swapping the text for the IDs in all the `env._` calls, then using PO files to map all of that to actual text. The mapping could be automated, but setting the IDs will require some work.

File: .github/ISSUE_TEMPLATE/Bug-Report.yml [hunk 1]
```diff
@@ -5,7 +5,7 @@ body:
   - type: markdown
     attributes:
       value: |
-        Thanks for taking the time to fill out this bug report for MCprep! We just need some info. Check here for more info: https://github.com/TheDuckCow/MCprep/issues/322. Please note that if you are looking for general help with Blender itself or a related problem like Mineways, you are better off joining the official [MCprep discord](https://discord.gg/mb8hBUC). **Note that support will not be provided if you're found to have an illegal copy of Minecraft!**
+        Thanks for taking the time to fill out this bug report for MCprep! We just need some info. Check here for more info: https://github.com/TheDuckCow/MCprep/issues/322. Please note that if you are looking for general help with Blender itself or a related problem like Mineways, you are better off joining the official [MCprep discord](https://discord.gg/mb8hBUC).
```
Comment (TheDuckCow) [inline .github/ISSUE_TEMPLATE/Bug-Report.yml:8]: Is there a reason for removing this recent addition here? (or just a mixup on syncing with dev?)

File: MCprep_addon/MCprep_resources/Languages/en/LC_MESSAGES/mcprep.po [hunk 2]
```diff
@@ -0,0 +1,523 @@
+# SOME DESCRIPTIVE TITLE.
+# Copyright (C) YEAR THE PACKAGE'S COPYRIGHT HOLDER
+# This file is distributed under the same license as the PACKAGE package.
+# FIRST AUTHOR <EMAIL@ADDRESS>, YEAR.
```
Comment (TheDuckCow) [inline MCprep_addon/MCprep_resources/Languages/en/LC_MESSAGES/mcprep.po:4]: We should probably fill these out/replace with the actual GPL header, or remove and have it just use the package's parent license inherently.

File: MCprep_addon/MCprep_resources/mcprep_data_update.json
Comment (TheDuckCow) [inline MCprep_addon/MCprep_resources/mcprep_data_update.json:1]: I tend to prefer to not edit this file during unrelated code changes, and rather make it its own change closer to the time of a release (even if it shows up as being modified locally). Unless is this just showing as a diff due to merging in dev? We could try changing the base branch and then changing it back to reset, weird trick that fixes the diffs

File: MCprep_addon/conf.py [hunk 3]
```diff
@@ -123,6 +125,13 @@ def __init__(self):
 		# that no reading has occurred. If lib not found, will update to [].
 		# If ever changing the resource pack, should also reset to None.
 		self.material_sync_cache = []
+		
+		# i18n using Python's gettext module
+		self.en_i18 = gettext.translation("mcprep", self.languages_folder, fallback=True, languages=['en'])
+
+		# We've made this a part of the env class 
+		# since it's probably better in the long run
+		self._ = self.en_i18.gettext
```
Comment (TheDuckCow) [inline MCprep_addon/conf.py:134]: This feels very odd to me, what is this line accomplishing?

File: MCprep_addon/conf.py [hunk 4]
```diff
@@ -210,6 +219,183 @@ def deprecation_warning(self):
 
 env = MCprepEnv()
 
+
+# ! Deprecated as of MCprep 3.5
+def init():
```
Comment (TheDuckCow) [inline MCprep_addon/conf.py:224]: Let's check diffs/make sure dev is merged in - I'll maybe hold off on further review/comments below until that's done

**Review by TheDuckCow — COMMENTED** [review]
Leaving just an initial comment status, as I think we need to refresh (either by flicking the base branch) or otherwise merging in the latest changes from dev. But still adding a few comments in the meantime. I like the idea of using the standardize python library, good to see that forming

File: .github/ISSUE_TEMPLATE/Bug-Report.yml (hunk 1 above)
Comment (StandingPadAnimations) [reply .github/ISSUE_TEMPLATE/Bug-Report.yml:8]: It's a mix up with dev syncing, I'm not sure how that happened

File: MCprep_addon/conf.py (hunk 3 above)
Comment (StandingPadAnimations) [reply MCprep_addon/conf.py:134]: It creates a functions called `_` in the env class. Generally `_` is used for localizing strings (in Python as well as C++) as it's pretty easy to type and recognizable by localization tools. However, I've put it in env since `_` is also syntax for a variable that's unused (like in a loop), and I'm not sure if that would be affected by making `_` global

File: .github/ISSUE_TEMPLATE/Bug-Report.yml (hunk 1 above)
Comment (TheDuckCow) [reply .github/ISSUE_TEMPLATE/Bug-Report.yml:8]: Cool - once resolve, just re-request review for me. Sorry it took ages to get to this in the first place

File: MCprep_addon/conf.py (hunk 3 above)
Comment (TheDuckCow) [reply MCprep_addon/conf.py:134]: Do you have some reference or examples in third party packages using _ for localization? More for my own learning. Definitely a fan of following convention, though want to make sure it remains clear as well. But it does make sense to use such short functions for things like that, since it'll be all over.

Comment (StandingPadAnimations) [reply MCprep_addon/conf.py:134]: The Python docs for [`gettext`](https://docs.python.org/3/library/gettext.html) use `_`:
```py
import gettext
gettext.bindtextdomain('myapplication', '/path/to/my/language/directory')
gettext.textdomain('myapplication')
_ = gettext.gettext
# ...
print(_('This is a translatable string.'))
```
This is also standard in the GNU gextext library for C (which is what the Python gettext is based on), and is the default keyword `xgettext` looks for:
https://en.m.wikipedia.org/wiki/Gettext#Operation

**Comment by StandingPadAnimations** [comment]
We're not taking translations at the moment, but submissions will be open when this is merged (hopefully by MCprep 3.6)

**Comment by StandingPadAnimations** [comment]
Ok I'm trying to refresh this branch, but GitHub is being a bit weird about it. Should we create a new branch and cherry-pick the changes on top?

**Comment by TheDuckCow** [comment]
If you need to force push, it's fine with me. Or a fresh branch, whatever you prefer to avoid less headache. As long as the changes are carried through that's the main thing (just be aware using a new branch won't correlate back to this PR)

**Comment by StandingPadAnimations** [comment]
Moved to #538
